From 485d3f5760d8feb8a5d842218c9cb980a78cc34d Mon Sep 17 00:00:00 2001
From: Oleg Nesterov <oleg@redhat.com>
Date: Sun, 5 Jul 2026 18:52:45 +0200
Subject: [PATCH 11566/15333] apparmor: use SEND_SIG_NOINFO instead of NULL in
 aa_audit()

SEND_SIG_NOINFO is defined as ((struct kernel_siginfo *) 0), so passing
NULL works, but:

- this works "by accident" and looks as if the caller doesn't understand
  the signal sending API.

- more importantly, this hides the usage of SEND_SIG_NOINFO from grep,
  and this is really bad.

Signed-off-by: Oleg Nesterov <oleg@redhat.com>
Acked-by: Georgia Garcia <georgia.garcia@canonical.com>
Signed-off-by: John Johansen <john.johansen@canonical.com>
---
 security/apparmor/audit.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/security/apparmor/audit.c b/security/apparmor/audit.c
index 4a60b6fda75f..15e42e96b163 100644
--- a/security/apparmor/audit.c
+++ b/security/apparmor/audit.c
@@ -192,7 +192,7 @@ int aa_audit(int type, struct aa_profile *profile,
 	aa_audit_msg(type, ad, cb);
 
 	if (ad->type == AUDIT_APPARMOR_KILL)
-		(void)send_sig_info(profile->signal, NULL,
+		send_sig_info(profile->signal, SEND_SIG_NOINFO,
 			ad->common.type == LSM_AUDIT_DATA_TASK &&
 			ad->common.u.tsk ? ad->common.u.tsk : current);
 
-- 
2.55.0

